capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -236,4 +244,76 @@ public static JsonObject createRequestBodyJson(HttpServletRequest request) throw
     }
     return retrievedObjects;
   }
-}
+
+  public static <T extends BaseEntity> T put(CollectionReference collectionReference, HttpServletRequest request, 
+      HttpServletResponse response, GenericClass<T> genericClass) throws IOException {
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    String id = "";
+    try {
+      id = jsonObject.get("id").getAsString();
+    } catch (Exception e) {
+      System.err.println("Error: " + e);
+      System.err.println("This error was likely caused by a lack of an \"id\" field in the post body.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    if (id.length() == 0) {
+      System.err.println("Error caused by either an empty or non-existent \"id\" field in the post body.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
+    Map<String, Object> update = new HashMap<>();
+    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    boolean containsParameter = false;
+    for (String key : jsonObject.keySet()) {
```

Comment:
Good idea, I'll change this.

Reply:
Actually on second thought, I think it makes sense to loop through the JSON object. I want to grab everything in the request body, and use that to update the object. The only way to do this is by looping through all of its keys. Otherwise, I won't notice if they send keys that are not object fields. And, they do not necessarily need to send every field to be updated, which causes this to inappropriately throw errors. I think I'm going to revert back to looping through the key set.